scenario/clickactionbutton.go

```diff
@@ -0,0 +1,643 @@
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/qlik-oss/enigma-go"
+	"github.com/qlik-oss/gopherciser/action"
+	"github.com/qlik-oss/gopherciser/connection"
+	"github.com/qlik-oss/gopherciser/enigmahandlers"
+	"github.com/qlik-oss/gopherciser/enummap"
+	"github.com/qlik-oss/gopherciser/senseobjdef"
+	"github.com/qlik-oss/gopherciser/session"
+)
+
+type (
+	// ClickActionButtonSettings implements the ActionSettings and
+	// ContainerAction interfaces. Executing this action replicate the
+	// behavour of clicking an "action-button" in Sense.
+	ClickActionButtonSettings struct {
+		// ID object id
+		ID string `json:"id" appstructure:"active:action-button" displayname:"Button ID" doc-key:"clickactionbutton.id"`
+	}
+)
+
+type (
+	// *** Sub actions of an actionButton ***
+	// An actionButton can contain multiple buttonActions and one buttonNavigationAction.
+	// These sub-actions are are executed in order when the an action-button is clicked.
+
+	buttonActionType int
+
+	buttonAction struct {
+		ActionLabel         string           `json:"actionLabel"`
+		ActionType          buttonActionType `json:"actionType"`
+		Bookmark            string           `json:"bookmark"`
+		Field               string           `json:"field"`
+		Variable            string           `json:"variable"`
+		ShowSystemVariables bool             `json:"showSystemVariables"`
+		SoftLock            bool             `json:"softLock"`
+		Value               string           `json:"value"`
+		CID                 string           `json:"cId"`
+	}
+
+	buttonNavigationActionType int
+
+	buttonNavigationAction struct {
+		Action     buttonNavigationActionType `json:"action"`
+		Sheet      string                     `json:"sheet"`
+		Story      string                     `json:"story"`
+		WebsiteURL string                     `json:"websiteUrl"`
+		SameWindow bool                       `json:"sameWindow"`
+	}
+)
+
+const (
+	emptyAction buttonActionType = iota
+	unknownAction
+	applyBookmark
+	moveBackwardsInSelections
+	moveForwardsInSelections
+	clearAllSelections
+	clearSelectionsInOtherFields
+	clearSelectionsInField
+	selectAllValuesInField
+	selectValuesInField
+	selectValuesMatchingSearchCriteria
+	selectAlternatives
+	selectExcluded
+	selectPossibleValuesInField
+	toggleFieldSelection
+	lockAllSelections
+	lockSpecificField
+	unlockAllSelections
+	unlockSpecificField
+	setVariableValue
+)
+
+const (
+	emptyNavAction buttonNavigationActionType = iota
+	noneNavAction
+	unknownNavAction
+	firstSheet
+	lastSheet
+	previousSheet
+	nextSheet
+	goToSheet
+	goToSheetByID
+	goToStory
+	openWebsite
+)
+
+// IsContainerAction implements ContainerAction interface
+// and sets container action logging to original action entry
+func (settings ClickActionButtonSettings) IsContainerAction() {}
+
+// Validate filter pane select action
+func (settings ClickActionButtonSettings) Validate() error {
+	if settings.ID == "" {
+		return errors.Errorf("Empty object ID")
```

Comment:
Would prefer to keep the error messages all lower-case as mentioned above (the current implementations are probably kinda mixed, we should go back and change..).